File: src/main/java/catwalks/util/meta/MetaProperty.java
package catwalks.util.meta;

/**
 * Created by TheCodeWarrior
 */
public abstract class MetaProperty {
	/**
	 * Returns true if the property supports bit extensions
	 */
	public abstract boolean addBits(int index, int bits);
	
	public abstract String getValue(MetaStorage storage);
}

File: src/main/java/catwalks/util/meta/MetaProperty.kt
package catwalks.util.meta

/**
 * Created by TheCodeWarrior
 */
abstract class MetaProperty {
    /**
     * Returns true if the property supports bit extensions
     */
    abstract fun addBits(index: Int, bits: Int): Boolean

    abstract fun getValue(storage: MetaStorage): String
}
